src/main/src/main/java/org/geoserver/config/GeoServerLoaderListener.java

```diff
@@ -0,0 +1,35 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.config;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.config.util.XStreamPersister;
+
+/**
+ * Listener interface for {@linkplain DefaultGeoServerLoader} load catalog and geoServer startup
+ * process.
+ */
+public interface GeoServerLoaderListener {
+
+    public static final GeoServerLoaderListener EMPTY_LISTENER = new EmptyGeoServerLoaderListener();
+
+    void loadCatalog(Catalog catalog, XStreamPersister xp);
```

Comment:
Javadoc missing. Is the even happening before or after loading?

Reply:
Indeed, added Javadoc describing it is running after loading.